<launch>
    <arg name="camera_namespace" default="zivid_camera"/>

    <group ns="$(arg camera_namespace)">
        <node name="zivid_camera" pkg="zivid_camera" type="zivid_camera_node" output="screen">
            <param name="serial_number" type="str" value=""/>
        </node>

        <node name="$(anon dynparam_general)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters settings">
            <param name="processing_filters_smoothing_gaussian_enabled" type="bool" value="true"/>
            <param name="processing_filters_smoothing_gaussian_sigma" type="double" value="1.0"/>
            <param name="processing_filters_reflection_removal_enabled" type="bool" value="true"/>
        </node>

        <node name="$(anon dynparam_f0)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters settings/acquisition_0">
            <param name="enabled" type="bool" value="true"/>
            <param name="brightness" type="double" value="1.0"/>
            <param name="exposure_time" type="int" value="40000"/>
            <param name="gain" type="double" value="1.0"/>
            <param name="aperture" type="double" value="5.66"/>
        </node>

        <node name="$(anon dynparam_f1)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters settings/acquisition_1">
            <param name="enabled" type="bool" value="true"/>
            <param name="brightness" type="double" value="1.0"/>
            <param name="exposure_time" type="int" value="40000"/>
            <param name="gain" type="double" value="1.0"/>
            <param name="aperture" type="double" value="3.67"/>
        </node>

        <node name="$(anon dynparam_f2)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters settings/acquisition_2">
            <param name="enabled" type="bool" value="true"/>
            <param name="brightness" type="double" value="1.0"/>
            <param name="exposure_time" type="int" value="40000"/>
            <param name="gain" type="double" value="1.0"/>
            <param name="aperture" type="doble" value="2.59"/>
        </node>
    </group>
</launch>
